script.js

```diff
@@ -1,12 +1,543 @@
+  let styleContainer = document.getElementById("styleContainer");
+  let footer = document.getElementById("footer");
+
+  if (styleContainer === null) {
+    styleContainer = document.createElement("div");
+    styleContainer.setAttribute("id", "styleContainer");
+    styleContainer.className = "style-container";
+    styleContainer.style.width = "98%";
+    styleContainer.style.flexWrap = "wrap";
+    styleContainer.style.margin = "0 auto";
+    rootElem.insertBefore(styleContainer, footer);
+  }
+  episodeList.forEach((episode) => {
+    styleContainer.appendChild(addSingleEpisode(episode));
+  });
+};
+
+function upDateDropDownMenu(episodeList) {
+  //goes through the options removes the options and recreates the options in the second loop
+  let select = document.getElementById("select");
+  if (select !== null) {
+    for (let index = select.options.length; index > 0; index--) {
+      select.remove(index);
+    }
+
+    episodeList.forEach((episode) => {
+      let option = document.createElement("option");
+      option.innerHTML = `${episode.name}: S${formatSeasonNum(
+        episode.season
+      )}E${formatSeasonNum(episode.number)}`;
+      select.appendChild(option);
+    });
+  }
+}
+
+const addDropDownMenu = (episodeList) => {
+  // select on the right  // checks if the drop down exist it creates it else it removes the options 222
+  let option;
+
+  // this condition prevents duplication
+  select = document.createElement("select");
+  select.style.display = "none"; // hide select episodes on the tv show listing
+
+  select.setAttribute("id", "select");
+  option = document.createElement("option");
+  option.innerText = "All episodes";
+  select.appendChild(option);
+
+  select.addEventListener("change", function (event) {
+    let episodeContainers = document.getElementsByClassName(
+      "episode-container-class-name"
+    ); // An array of episode container
+
+    let episodeCounter = 0;
+    let containers = Array.from(episodeContainers);
+
+    if (event.target.options[event.target.selectedIndex].index === 0) {
+      // refers to the all episode option
+      containers.forEach((container) => {
+        episodeCounter = containers.length;
+        container.style.display = "";
+      });
+    } else {
+      containers.forEach((container, episodeIndex) => {
+        if (
+          episodeIndex ===
+          event.target.options[event.target.selectedIndex].index - 1 // test if the current index is equal to the index
+        ) {
+          showEpisode(container); //.style.display = "";
+        } else {
+          hideEpisode(container); //.style.display = "none";
+        }
+        // This increments until it matches the episode index});
+      });
+      episodeCounter = 1; // It keeps counting until it reaches the end of the array
+    }
+    updateCounter(episodeCounter, containers.length);
+  });
+  return select;
+};
+
+// counter function display creates the dom
+const addCounter = (_) => {
```

Comment:
You don't need to pass a parameter to an arrow function. Can be just written as `const addCounter = () => {`